static/js/server_events_dispatch.js

```diff
@@ -286,6 +288,15 @@ export function dispatch_normal_event(event) {
                                     );
                                 }
                             }
+                            if (user_settings.realm_icon_as_favicon) {
+                                // it's a bit strange that we're calling a function in
+                                // notifications to redraw the favicon... but we do this
+                                // because favicon.js does not keep track of unreads,
+                                // once they have been rendered.
+                                notifications.redraw_favicon();
+                                // incase we fail to render the svg
```

Comment:
Can you explain this possibility in more detail?

Also, should `notifications.redraw_favicon` just be responsible for doing `favicon.update_attrs_and_alt_icon` instead?